<?xml version="1.0" encoding="UTF-8"?>
<!-- File Renamer is a desktop app for Linux and Windows for batch renaming files. -->
<!-- https://github.com/flossapps/file-renamer -->
<!-- Copyright (C) 2024 Carlos, GNU General Public License -->
<!-- https://www.gnu.org/licenses/gpl-3.0.html -->
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>File Renamer</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>APP</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="browse">
     <property name="focusPolicy">
      <enum>Qt::FocusPolicy::StrongFocus</enum>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="filter_label">
        <property name="text">
         <string>Filter</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="filter_txt">
        <property name="text">
         <string>*.*</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="dir_btn">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="dir_txt">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QTextEdit" name="dir_output">
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QGroupBox" name="options">
     <property name="minimumSize">
      <size>
       <width>788</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QCheckBox" name="recursively">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Recursively</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="extension">
        <property name="toolTip">
         <string/>
        </property>
        <property name="text">
         <string>Keep Extension</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="path">
        <property name="minimumSize">
         <size>
          <width>160</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Path</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="id">
        <property name="minimumSize">
         <size>
          <width>160</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Keep ID</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="sort">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>160</width>
          <height>0</height>
         </size>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LayoutDirection::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Sort</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string/>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QComboBox" name="comboBox">
        <item>
         <property name="text">
          <string>Select</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Remove Non-Alphanum Chars</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Remove Accents</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Trim Spaces</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Replace Spaces With Hyphens</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Replace Dots With Hyphens</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Replace Hyphens With Spaces</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Lower Case</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Title Case</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Remove ID</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Number</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="clear_btn">
        <property name="text">
         <string>CLEAR</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="rename_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>RENAME</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QGroupBox" name="search_replace">
     <property name="title">
      <string/>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="0">
       <widget class="QLabel" name="search_label">
        <property name="text">
         <string>Search</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="search">
        <property name="tabletTracking">
         <bool>false</bool>
        </property>
        <property name="focusPolicy">
         <enum>Qt::FocusPolicy::StrongFocus</enum>
        </property>
        <property name="text">
         <string notr="true"/>
        </property>
        <property name="placeholderText">
         <string notr="true"/>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="replace_label">
        <property name="text">
         <string>Replace</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLineEdit" name="replace">
        <property name="tabletTracking">
         <bool>false</bool>
        </property>
        <property name="focusPolicy">
         <enum>Qt::FocusPolicy::StrongFocus</enum>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QCheckBox" name="regex">
        <property name="toolTip">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Noto Sans'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Google Python Regex Examples&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Regular Expression</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="find_btn">
        <property name="text">
         <string>Find</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>dir_btn</tabstop>
  <tabstop>dir_txt</tabstop>
  <tabstop>dir_output</tabstop>
  <tabstop>recursively</tabstop>
  <tabstop>id</tabstop>
  <tabstop>extension</tabstop>
  <tabstop>comboBox</tabstop>
  <tabstop>clear_btn</tabstop>
  <tabstop>rename_btn</tabstop>
  <tabstop>search</tabstop>
  <tabstop>replace</tabstop>
  <tabstop>regex</tabstop>
  <tabstop>find_btn</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>dir_btn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>open_dir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>457</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dir_btn</sender>
   <signal>clicked()</signal>
   <receiver>dir_txt</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>463</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>686</x>
     <y>78</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_btn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>Widget</receiver>
   <slot>index_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>548</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rename_btn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rename_files()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>778</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>794</x>
     <y>553</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>recursively</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>add_recursively()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>491</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>id</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>keep_id()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>797</x>
     <y>484</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>extension</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>keep_ext()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>287</x>
     <y>465</y>
    </hint>
    <hint type="destinationlabel">
     <x>794</x>
     <y>526</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>regex</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>regex()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>565</x>
     <y>577</y>
    </hint>
    <hint type="destinationlabel">
     <x>33</x>
     <y>602</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>find_btn</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>find()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>726</x>
     <y>555</y>
    </hint>
    <hint type="destinationlabel">
     <x>776</x>
     <y>595</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sort</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>sort()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>731</x>
     <y>455</y>
    </hint>
    <hint type="destinationlabel">
     <x>801</x>
     <y>427</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>path</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>path()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>797</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open_dir()</slot>
  <slot>index_changed()</slot>
  <slot>rename_files()</slot>
  <slot>add_recursively()</slot>
  <slot>keep_ext()</slot>
  <slot>keep_id()</slot>
  <slot>search_replace()</slot>
  <slot>regex()</slot>
  <slot>find()</slot>
  <slot>clear()</slot>
  <slot>sort()</slot>
  <slot>path()</slot>
 </slots>
</ui>
